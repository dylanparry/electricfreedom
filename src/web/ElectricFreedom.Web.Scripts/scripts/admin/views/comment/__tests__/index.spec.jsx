import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import CommentSwitch from '../index';
import CommentList from '../list';

const setup = (propOverrides) =>
{
  const wrapper = shallow(<CommentSwitch />);

  const props = Object.assign({
    initialEntries: ['/'],
  }, propOverrides);
  const mounted = mount((
    <MemoryRouter {...props}>
      <CommentSwitch />
    </MemoryRouter>
  ));

  return {
    wrapper,
    mounted,
  };
};

describe('CommentSwitch', () =>
{
  test('render', () =>
  {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('route "/comments" renders CommentList', () =>
  {
    const { mounted } = setup({
      initialEntries: ['/comments'],
    });

    expect(mounted.find(CommentList).exists()).toBe(true);
  });
});
