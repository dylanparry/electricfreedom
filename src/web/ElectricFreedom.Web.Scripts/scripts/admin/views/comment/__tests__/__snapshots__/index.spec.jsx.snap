// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentSwitch render 1`] = `
<commentSwitch>
  <Switch />
</commentSwitch>
`;
